tests/integration/test_prometheus_integration.py

```diff
@@ -113,3 +113,88 @@ async def test_prometheus_integration(
         await asyncio.sleep(5)
     else:
         pytest.fail("Metric didn't get to Prometheus")
+
+
+@pytest.mark.abort_on_fail
+async def test_prometheus_ca_integration(
+    ops_test: OpsTest,
+    pushgateway_charm: Path,
+    tester_charm: Path,
+):
+    """Validate the integration between the Pushgateway and Prometheous."""
+    prometheus_app_name = "prometheus"
+    tester_name = "testingcharm"
+    ca_name = "ca"
+    apps = [APP_NAME, prometheus_app_name, tester_name, ca_name]
+
+    image = METADATA["resources"]["pushgateway-image"]["upstream-source"]
+    resources = {"pushgateway-image": image}
+
+    await asyncio.gather(
+        ops_test.model.deploy(
+            pushgateway_charm,
+            resources=resources,
+            application_name=APP_NAME,
+        ),
+        ops_test.model.deploy(
+            "prometheus-k8s",
```

Comment:
Prom is deployed here again without cleaning up the model from the previous test method.
But instead cleanup, perhaps move to a separate file?